### ISSUE TITLE
cannot unpack non-iterable NoneType object with `SOFT_DELETE_CASCADE`

### ISSUE THREAD

**Issue by hamidrabedi** [makinacorpus/django-safedelete#235]
I have a parent child model and i want if the parent object got deleted all the children get deleted too (safe delete in my case):

```
class Review(SafeDeleteModel):
    _safedelete_policy = SOFT_DELETE_CASCADE
    parent = models.ForeignKey(
        'self',
        verbose_name=_("Parent"),
        related_name="children",
        on_delete=models.CASCADE,
        null=True,
        blank=True
    )
```

and if i delete the parent object this error will be raised:
`cannot unpack non-iterable NoneType object`
![image](https://github.com/makinacorpus/django-safedelete/assets/92696830/0d865363-c1ef-48ce-818d-3d7358caea13)

**Comment by Gagaro** [comment]
I just added a test with your model which works: https://github.com/makinacorpus/django-safedelete/commit/6cdd3e8d079896d61efca22c6abe51f10ce26705

Are you using the last version of safedelete? Do you overwrite the delete method?

**Comment by razdva0** [comment]
I got the same error on macbook air m1

**Comment by sazzouz** [comment]
Same issue 1.3.2, haven't overwritten any methods, occurring when using the SOFT_DELETE_CASCADE policy. Could it be to do with handling cascaded soft deletions when there are no related objects?

**Comment by elihuihms** [comment]
Same issue with 1.3.3, no overridden `.delete` methods.

**Comment by hamidrabedi** [comment]
@Gagaro No, it seems like soft delete has problems with children-parent models, also has with django-mptt for tree models. related issue in django-mptt package:
https://github.com/django-mptt/django-mptt/issues/745

**Comment by Gagaro** [comment]
I made a test that passed without issue. Are you using your model like that or do you have any idea how to reproduce it?

https://github.com/makinacorpus/django-safedelete/commit/6cdd3e8d079896d61efca22c6abe51f10ce26705#diff-985ae0f484f9a15bfb9b8e016c13075eb6896094ff89516544436dce6c63c4feR297

**Comment by hamidrabedi** [comment]
@Gagaro Try deleting from django admin.

**Comment by razdva0** [comment]
My friend overwritten delete method without return like this:
`
def delete(self, force_policy=None, **kwargs):
    ...
    super().delete(force_policy, **kwargs)
`
everything is ok with return

**Comment by elihuihms** [comment]
> My friend overwritten delete method without return like this: `def delete(self, force_policy=None, **kwargs): super().delete(force_policy, **kwargs)` everything is ok with return

To expand on @razdva0 answer, I had a model with an overridden `delete()` method that did not return the tuple returned by `SafeDeleteModel.delete()`; that's what generated this error. When I captured and returned the tuple the error went away.

**Comment by Gagaro** [comment]
Thanks for the feedbacks. @hamidrabedi , do you have a delete method which does not return anything?

**Comment by hamidrabedi** [comment]
@Gagaro Yes i do have a delete method, but it calls the super method:

```py
    def delete(self, *args, **kwargs):
        """
        This method calls the superclass's delete()
        method with the given parameters, and then invalidates
        the cache for the query set's model.
        """
        result = super().delete(**kwargs)
        self.invalidate_cache()
        return result
```

**Comment by Gagaro** [comment]
Do you have more parent classes? Do you have a value for `result` if you try to print it?

**Comment by hamidrabedi** [comment]
@Gagaro I found where the problem is, when i try to delete one object normally everything works just fine, but when i call the queryset.delete method in the admin, it raises the error:
![image](https://github.com/makinacorpus/django-safedelete/assets/92696830/89528254-9600-4a1b-b2fc-c29621d83d69)

what gets called in django admin
![image](https://github.com/makinacorpus/django-safedelete/assets/92696830/7fea98fa-071b-4ea2-9486-04eb5801ab35)

So when you delete one object there is no response returning and it tries to unpack it :
![image](https://github.com/makinacorpus/django-safedelete/assets/92696830/9bf5287c-45f6-455f-b2fa-2f9242092f23)

**Comment by hamidrabedi** [comment]
> Do you have more parent classes? Do you have a value for `result` if you try to print it?

there is no result it raises errors

**Comment by Gagaro** [comment]
What classes are you inheriting from?

**Comment by hamidrabedi** [comment]
@Gagaro 
`from safedelete.admin import SafeDeleteAdmin`
`from safedelete.models import SafeDeleteModel`

**Comment by Gagaro** [comment]
It looks good to me. Could you maybe step into all the subsequent `delete()` to be sure there isn't one which does not return anything?

**Comment by hamidrabedi** [comment]
@Gagaro Oh i see, delete got overridden and is returning data.ok i got it.
